### PR TITLE
Endpoints additional info

### PR DESCRIPTION
**Pull request by GeneralSnuggles**
added last_scanned, and column names (Hostname, IP, MAC)

### CODE REVIEW

File: views/endpoints.ejs [hunk 1]
```diff
@@ -2,29 +2,52 @@
...
                     </p>
                     <p class="card-text">
-                        <%= endpoint.primary_mac %>
+                        <strong>Uptime:</strong> <%= (endpoint.uptime / (1000 * 60 * 60 * 24)).toFixed(2) %> days
+                    </p>
+                    <p class="card-text">
+                        <strong>Last Seen:</strong> <%= new Date(endpoint.seen_time).toLocaleString() %>
+                    </p>
+                    <p class="card-text">
+                        <strong>Last Scan:</strong> <%= endpoint.last_execution ? new Date(endpoint.last_execution).toLocaleString() : 'N/A' %>
```
Comment (GeneralSnuggles) [inline views/endpoints.ejs]: @maziland here should be replaced by data from the Scripts API
Do you want to check what I suggested on https://biosquery.atlassian.net/browse/BQU-16?

Comment (maziland) [reply views/endpoints.ejs]: Looks good

**Comment by maziland** [comment]
Is it ready for merging? @GeneralSnuggles

**Comment by GeneralSnuggles** [comment]
No.

**Comment by GeneralSnuggles** [comment]
Yes.

File: services/fleet.js [hunk 2]
```diff
@@ -62,13 +60,47 @@ async function getScriptByEndpoint(endpointList){
...
     return allScripts;
+}
+
+const mergeEndpointAndScripts = (endpoints, scriptsData) => {
+    // Ensure endpoints is an array
+    if (!Array.isArray(endpoints)) {
+        throw new TypeError('Expected endpoints to be an array');
+    }
+
+    console.log('Scripts Data:', scriptsData); // Debug output
```
Comment (maziland) [inline services/fleet.js]: Is it necessary? it prints out a lot of lines and makes it hard to watch for error

File: public/js/endpoint.js
Comment (maziland) [inline public/js/endpoint.js:1]: I'd move this whole function to a file in `public/js/utils.js`.
It has nothing to do with endpoint, it's more of a helper function

File: controllers/endpoints.js [hunk 3]
```diff
@@ -1,24 +1,39 @@
...
+            filteredEndpoints = endpointsWithScripts.filter(host => {
                 return host.hostname.toLowerCase().includes(searchQuery) ||
                     host.primary_ip.toLowerCase().includes(searchQuery) ||
                     host.primary_mac.toLowerCase().includes(searchQuery);
             });
         }
-
-        res.render("endpoints.ejs", { endpoints: endpoints.hosts });
+        filteredEndpoints.forEach(endpoint => {
+            endpoint.formatted_last_scan = formatDate.formatDate(endpoint.last_scan);
```
Comment (maziland) [inline controllers/endpoints.js]: You are using a function which is on the `public/js` path, which belongs to the frontend.
It is not the best practice, you should move it to be backend only/frontend only

File: controllers/endpoints.js [hunk 4]
```diff
@@ -1,24 +1,39 @@
 const fleetService = require('../services/fleet');
+const formatDate = require('../public/js/endpoint.js');
```
Comment (maziland) [inline controllers/endpoints.js]: Not recommended. Frontend <--> Backend imports == 💥

**Review by maziland — CHANGES_REQUESTED** [review]

File: services/fleet.js (hunk 2 above)
Comment (GeneralSnuggles) [reply services/fleet.js]: deleted

File: public/js/endpoint.js
Comment (GeneralSnuggles) [reply public/js/endpoint.js:1]: changed

File: controllers/endpoints.js (hunk 3 above)
Comment (GeneralSnuggles) [reply controllers/endpoints.js]: moved to the same endpoint.js (instead of public/js/enpoint.js)

File: controllers/endpoints.js (hunk 4 above)
Comment (GeneralSnuggles) [reply controllers/endpoints.js]: removed - no longer needed

**Review by maziland — APPROVED** [review]

**Pull request merged**